live-examples/js-examples/string/string-iterator.html

```diff
@@ -0,0 +1,13 @@
+<pre>
+<code id="static-js">const str = "The quick red fox jumped over the lazy dog's back.";
+
+let iterator = str[Symbol.iterator]();
+let theChar = iterator.next();
+
+while(!theChar.done) {
+  // output each character in the string, one per line
+  console.log(theChar);
+  theChar = iterator.next();
+}  
```

Comment:
It feels a bit odd to use a while loop here - in this situation (where we want to iterate through every value) it would be more natural to use `for...of`. But I understand the value of showing `next()`, `done` and so on.

What do you think about something like:

```
while(!theChar.done && theChar.value !== ' ') {
  // output each character in the first word, one per line
  console.log(theChar.value);
  theChar = iterator.next();
} 
```

...this is a slightly less natural fit for `for...of`, because you'd have to break out of the loop. What do you think?

Reply:
Made the change. I know it seemed contrived, but I wanted to make it clear what was going on.